onebusaway-android/src/main/java/org/onebusaway/android/provider/ObaProvider.java

```diff
@@ -225,6 +225,9 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                                 ObaContract.RegionOpen311Servers.API_KEY + " VARCHAR NOT NULL, " +
                                 ObaContract.RegionOpen311Servers.BASE_URL + " VARCHAR NOT NULL " +
                                 ");");
+                db.execSQL(
```

Comment:
This `ALTER TABLE`  statement needs to be moved to its own:

`if (oldVersion == 23) {`

...IF block, and a `++oldVersion;` needs to be added to the end of the `if (oldVersion == 22) {` IF block.

Otherwise the database won't upgrade properly.